lua/luasnip/extras/_treesitter.lua

```diff
@@ -0,0 +1,476 @@
+local util = require("luasnip.util.util")
+local tbl = require("luasnip.util.table")
+
+local function get_lang(bufnr)
+	local ft = vim.api.nvim_buf_get_option(bufnr, "ft")
+	local lang = vim.treesitter.language.get_lang(ft) or ft
+	return lang
+end
+
+-- Inspect node
+---@param node TSNode?
+---@return string
+local function inspect_node(node)
+	if node == nil then
+		return "nil"
+	end
+
+	local start_row, start_col, end_row, end_col =
+		vim.treesitter.get_node_range(node)
+
+	return ("%s [%d, %d] [%d, %d]"):format(
+		node:type(),
+		start_row,
+		start_col,
+		end_row,
+		end_col
+	)
+end
+
+---@param bufnr number
+---@param match string
+---@return LanguageTree, string
+local function reparse_buffer_after_removing_match(bufnr, match)
```

Comment:
I think we should make this function a bit more modular:
It only gets the text-range it should remove from the buffer-text, not the `match` and then has to retrieve the cursor to find the correct position

Reply:
Make sense! Update it!